rs/sns/governance/canister/governance.did

```diff
@@ -21,6 +21,135 @@ type Action = variant {
   ExecuteGenericNervousSystemFunction : ExecuteGenericNervousSystemFunction;
   ManageLedgerParameters : ManageLedgerParameters;
   Motion : Motion;
+  RegisterExtension : RegisterExtension;
+  DeregisterExtension : DeregisterExtension;
+  ExecuteExtensionFunction : ExecuteExtensionFunction;
+};
+
+type ExistingCanister = record {
+  canister_id : opt principal;
+  wasm_module_hash : opt blob;
+};
+
+type NewCanister = record {
+  chunked_canister_wasm : opt ChunkedCanisterWasm;
+};
+
+type Installation = variant {
+  existing_canister : ExistingCanister;
+  new_canister : NewCanister;
+};
+
+type Icrc2Token = variant {
+  NativeSnsToken : record {};
+  Icp : record {};
+};
+
+type Icrc2Allowance = record {
+    token : opt Icrc2Token;
+    amount_e8s : opt nat64;
+};
+
+type Allowances = record {
+  icrc2_allowances : opt vec Icrc2Allowance;
+};
+
+// Used in responses, e.g., list_nervous_system_functions.
+type ExtensionType = variant {
+  TreasuryManager;
+};
+
+// Used in requests, e.g., manage_neuron (to submit a RegisterExtension proposal).
+type TreasuryManagerInit = record {
+  // The initial allowances for the Treasury Manager.
+  initial_allowances : opt Allowances;
+};
+
+type ExtensionInit = variant {
+  TreasuryManagerInit : TreasuryManagerInit;
+};
+
+// Native proposal type: RegisterExtension.
+type RegisterExtension = record {
+  // Determines the type of installation for the extension canister; e.g., whether it is
+  // a new canister to be installed by the SNS or an existing one.
+  installation : opt Installation;
```

Comment:
I think we should specify the wasm hash, and maybe where to get the wasm?  For smaller wasms, it could be included as part of the proposal.

Reply:
> we should specify the wasm hash

That is already the case now, it's inside `Installation`.

> where to get the wasm

This is part of the metadata that is not up to the client but rather for the SNS to fetch, as we should only allow known extensions from the NNS-curated list (for which all metadata are already publicly known).